paddle/fluid/operators/sum_mkldnn_op.cc

```diff
@@ -236,5 +236,8 @@ class SumMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
 }  // namespace operators
 }  // namespace paddle
 
-REGISTER_OP_KERNEL(sum, MKLDNN, ::paddle::platform::CPUPlace,
-                   paddle::operators::SumMKLDNNOpKernel<float>);
+REGISTER_OP_KERNEL(
+    sum, MKLDNN, ::paddle::platform::CPUPlace,
+    paddle::operators::SumMKLDNNOpKernel<float>,
+    paddle::operators::SumKernel<paddle::platform::CPUDeviceContext, int>,
+    paddle::operators::SumKernel<paddle::platform::CPUDeviceContext, int64_t>);
```

Comment:
Could you tell us the background of this issue, why do you add int and int64 for sum_op?

Reply:
@luotao1 This fall back to a naive version of the code is needed because a type of the input data has the `int` type. Additionally the MKLDNN framework does not support the `int64` type. This problem is visible in the crn-ctc model - during the training.

Reply:
Can we workaround the issue you mentioned about crn-ctc model?

Because they are already registered on Plain,
```
REGISTER_OP_CPU_KERNEL(
    sum, ops::SumKernel<paddle::platform::CPUDeviceContext, float>,
    ops::SumKernel<paddle::platform::CPUDeviceContext, double>,
    ops::SumKernel<paddle::platform::CPUDeviceContext, int>,
    ops::SumKernel<paddle::platform::CPUDeviceContext, int64_t>);
```

It seems do not make sense we register in MKLDNN with duplicated function. And we have too many registered op now, I am not sure if this would slow down the searching time.

Anyway, the best choice is we could find a way workaround the specific issue of crn-ctc model.